wfe2/wfe.go

```diff
@@ -1273,3 +1276,40 @@ func (wfe *WebFrontEndImpl) NewOrder(
 		return
 	}
 }
+
+// Order is used to retrieve a existing order object
+func (wfe *WebFrontEndImpl) Order(ctx context.Context, logEvent *requestEvent, response http.ResponseWriter, request *http.Request) {
+	id, err := strconv.ParseInt(request.URL.Path, 10, 64)
```

Comment:
Let's make the registration ID part of the order path, and check it against the `.RegistrationID` of the returned error. This may give us some flexibility on choosing different indexing and/or partitioning schemes in the future.

Reply:
Presumably `s/error/order/` here?

Reply:
Also how do you imagine it being structured, `/acme/order/{reg-id}/{order-id}` or `/acme/reg/{reg-id}/orders/{order-id}` or something else?

Reply:
yep, s/error/order/ :-)

I was thinking /acme/order/{reg-id}/{order-id}

Reply:
If you get the order ID right but the reg ID wrong would it be 404 or 401?

Reply:
404